src/bulk/common.ts

```diff
@@ -1260,39 +1261,44 @@ export abstract class BulkOperationBase {
     options?: BulkWriteOptions | Callback<BulkWriteResult>,
     callback?: Callback<BulkWriteResult>
   ): Promise<BulkWriteResult> | void {
-    if (typeof options === 'function') (callback = options), (options = {});
-    options = options ?? {};
-
-    if (this.s.executed) {
-      return handleEarlyError(new MongoBatchReExecutionError(), callback);
-    }
+    callback =
+      typeof callback === 'function'
+        ? callback
+        : typeof options === 'function'
+        ? options
+        : undefined;
+
+    return maybeCallback(async () => {
+      options = options != null && typeof options !== 'function' ? options : {};
```

Comment:
can this be moved out of `maybeCallback`?

Reply:
Not without casting away a TS error, so this is preferable, see other reply